packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js

```diff
@@ -473,6 +480,113 @@ export const RemoteServerFactory = {
         })
     },
 
+    /**
+     * @private
+     */
+    _setupRemoveBasePathFromPathMiddleware(app) {
+        // Use envBasePath as the feature flag for this middleware
+        // If envBasePath is `/`, '', or undefined when the server starts, we don't need to
+        // initialize this middleware.
+        if (getEnvBasePath()) {
+            const removeBasePathFromPath = (path) => {
+                const regex = new RegExp(`^${getEnvBasePath()}(/|$)`)
+                return path.replace(regex, '/')
+            }
+
+            const _convertExpressRouteToRegex = (routePattern) => {
+                if (!routePattern) return null
+                if (routePattern instanceof RegExp) return routePattern
+                if (typeof routePattern !== 'string') return null
+
+                // Replace route parameters like :id with regex capture groups
+                let regexPattern = routePattern
+                    .replace(/:[^/]+/g, '[^/]+')
+                    .replace(/\/\*/g, '/.*')
+                    .replace(/\*/g, '.*')
+
+                // Escape other regex special characters except those we just handled
+                regexPattern = regexPattern.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')
+
+                // Unescape the patterns we want to keep
+                regexPattern = regexPattern
+                    .replace(/\\\[\\\^\/\\\]\\\+/g, '[^/]+')
+                    .replace(/\\\/\\\.\\\*/g, '/.*')
+                    .replace(/\\\.\\\*/g, '.*')
+                    .replace(/\\\(/g, '(')
+                    .replace(/\\\)/g, ')')
+                    .replace(/\\\?/g, '?')
+
+                return new RegExp(`^${regexPattern}$`)
+            }
+
+            /**
+             * Very early request processing.
+             *
+             * If the server receives a request containing the base path, remove it before allowing it through
+             *
+             * @param req {express.req} the incoming request - modified in-place
+             * @private
+             */
+            const removeBasePathFromPathMiddleware = (req, res, next) => {
+                // Scope base path removal to /mobify routes and routes defined by the express app (ie. worker.js)
+                // This is to avoid affecting other paths where a base path might be present if it happens to
+                // be equal to a site id.
+                // For example, if you have a base path of /us and a site id of /us we don't want
+                // to remove the /us from www.example.com/us/en-US/category/...
+
+                const basePath = getEnvBasePath()
+                let shouldRemoveBasePath = false
+
+                if (basePath) {
```

Comment:
what if `basePath` is just '/' is that a valid string? Should we make sure it is a valid string instead of here

Reply:
There is code inside the `getEnvBasePath()` function that handles the case of envBasePath being set to `/`; where the `/` is treated as though you set envBasePath to an empty string.

The code inside getEnvBasePath will also handle non-string values and return an empty string if that is the case.